answerking_app/utils/mixins/CategoryProductMixins.py

```diff
@@ -0,0 +1,21 @@
+from django.db.models import QuerySet
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from answerking_app.models.models import Product
+from answerking_app.models.serializers import ProductSerializer
+from answerking_app.utils.mixins.ApiExceptions import HttpErrorResponse
+
+
+class CategoryProductListMixin:
+    def list(self, request: Request, cat_id: int, *args, **kwargs) -> Response:
+
+        products: QuerySet[Product] = Product.objects.filter(
+            category__id=cat_id
```

Comment:
should be single underline not double for category_id

Reply:
This is deliberate it's needed for the filtering